src/vivarium/framework/results/context.py

```diff
@@ -101,7 +101,9 @@ def add_observation(
             (name, aggregator_sources, aggregator, additional_keys)
         )
 
-    def gather_results(self, population: pd.DataFrame, event_name: str) -> Dict[str, float]:
+    def gather_results(
+        self, population: pd.DataFrame, event_name: str
+    ) -> Generator[Dict[str, float], None, None]:
```

Comment:
What do the Nones here mean?

Reply:
@stevebachmeier I'm assuming this is why we saw the "expected 3 arguments but got only 1" at mid-day yesterday? I remember reading this while looking into it: https://stackoverflow.com/questions/42531143/how-to-type-hint-a-generator-in-python-3

Reply:
So I won't pretend to have a good grasp on how generators work, but the return types of a generator are [yield, send, and return]. We're yielding, so that's the one we want. I don't even know what the other two mean at this point, though.

https://stackoverflow.com/questions/42531143/how-to-type-hint-a-generator-in-python-3